pkg/operator/controllers/autosizednodes/autosizednodes_controller.go

```diff
@@ -0,0 +1,126 @@
+	"github.com/coreos/ignition/v2/config/util"
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers"
+)
+
+type Reconciler struct {
+	client.Client
+	Scheme *kruntime.Scheme
+
+	log *logrus.Entry
+}
+
+const (
+	configName = "dynamic-node"
+)
+
+func NewReconciler(log *logrus.Entry, mgr ctrl.Manager) *Reconciler {
+
+	return &Reconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+
+		log: log,
+	}
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	var aro arov1alpha1.Cluster
+	var config mcv1.KubeletConfig
+	var err error
+
+	err = r.Get(ctx, request.NamespacedName, &aro)
+	if err != nil {
+		r.log.Error("unable to fetch aro cluster")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	r.log.Infof("Config changed, autoSize: %t\n", aro.Spec.Features.ReconcileAutoSizedNodes)
+
+	// key is used to locate the object in the etcd
+	key := types.NamespacedName{
+		Name: configName,
+	}
+
+	err = r.Get(ctx, key, &config)
+	if client.IgnoreNotFound(err) != nil {
+		r.log.Errorf("could not fetch KubeletConfig: %v", err)
+		return ctrl.Result{RequeueAfter: time.Hour}, err
+	}
+
+	if kerrors.IsNotFound(err) {
+		if aro.Spec.Features.ReconcileAutoSizedNodes {
+			//create
+			config := makeConfig()
+
+			err = r.Create(ctx, config, &client.CreateOptions{})
+			if err != nil {
+				r.log.Errorf("could not create KubeletConfig: %v", err)
+			}
+			return ctrl.Result{}, err
+		}
+	} else {
+		if !aro.Spec.Features.ReconcileAutoSizedNodes {
+			//remove
+			if err == nil && config.Spec.AutoSizingReserved != nil && *config.Spec.AutoSizingReserved {
+				// the right config is there delete
+				err = r.Delete(ctx, &config, &client.DeleteOptions{})
+				if err != nil {
+					r.log.Errorf("could not delete KubeletConfig: %v", err)
+				}
+				return ctrl.Result{}, client.IgnoreNotFound(err)
+			}
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager prepares the controller with info who to watch
+// TODO: Add informers with a cache to reduce the amount of API calls
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	builder := ctrl.NewControllerManagedBy(mgr).For(&arov1alpha1.Cluster{})
+
+	return builder.
+		Named(controllers.AutoSizedNodesControllerName).
+		Owns(&mcv1.KubeletConfig{}).
+		Complete(r)
+}
+
+func makeConfig() *mcv1.KubeletConfig {
+	return &mcv1.KubeletConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "machineconfiguration.openshift.io/v1",
+			Kind:       "KubeletConfig",
+		},
```

Comment:
When you use a versioned client with a scheme set up properly - you don't have to add `TypeMeta`. It is being added automatically. Is it not the case with the split client?